File: android/apollo/src/main/java/io/muun/apollo/data/net/base/interceptor/LanguageHeaderInterceptor.java
package io.muun.apollo.data.net.base.interceptor;

import io.muun.apollo.data.net.base.BaseInterceptor;
import io.muun.apollo.domain.utils.ExtensionsKt;
import io.muun.common.net.HeaderUtils;

import android.content.Context;
import okhttp3.Request;

import javax.inject.Inject;

public class LanguageHeaderInterceptor extends BaseInterceptor {

    private final Context applicationContext;

    @Inject
    public LanguageHeaderInterceptor(Context applicationContext) {
        this.applicationContext = applicationContext;
    }

    @Override
    protected Request processRequest(Request request) {

        final String lang = ExtensionsKt.locale(applicationContext).getLanguage();

        return request.newBuilder()
                .addHeader(
                        HeaderUtils.CLIENT_LANGUAGE,
                        !lang.isEmpty() ? lang : HeaderUtils.DEFAULT_LANGUAGE_VALUE
                )
                .build();
    }
}

File: android/apollo/src/main/java/io/muun/apollo/data/net/base/interceptor/LanguageHeaderInterceptor.kt
package io.muun.apollo.data.net.base.interceptor

import android.content.Context
import io.muun.apollo.data.net.base.BaseInterceptor
import io.muun.apollo.domain.utils.locale
import io.muun.common.net.HeaderUtils
import okhttp3.Request
import javax.inject.Inject

class LanguageHeaderInterceptor @Inject constructor(
    private val applicationContext: Context,
) : BaseInterceptor() {

    override fun processRequest(originalRequest: Request): Request {
        val language = applicationContext.locale().language
        return originalRequest.newBuilder()
            .addHeader(
                HeaderUtils.CLIENT_LANGUAGE,
                language.ifEmpty { HeaderUtils.DEFAULT_LANGUAGE_VALUE }
            )
            .build()
    }
}